Jam Tools/Source/HTTPServer.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+  ==============================================================================
+
+    HTTPServer.cpp
+    Created: 19 Aug 2018 10:46:25pm
+    Author:  Matthias Pueski
+
+  ==============================================================================
+*/
+
+#include "HTTPServer.h"
+#include "../JuceLibraryCode/JuceHeader.h"
+#include "MessageBus/MessageBus.h"
+
+juce::String makeResponse(juce::String message) {
+    juce::String header = "";
+    header += "HTTP/1.1 200 OK\r\n";
+    header += "Content-Type:text/html\r\n";
+
+    juce::String body = "";
+    body += "<html><head><title>Jam App</title></head><body><h1>Jam App</h1><p>";
+    body += message;
+    body += "</p></body></html>\r\n";
+
+    header += "Content-Length:"+juce::String(body.length())+"\r\n\r\n";
+
+    return header + body;
+}
+
+HTTPServer::HTTPServer() : juce::Thread("Server") {
+
+}
+
+HTTPServer::HTTPServer(int port) : juce::Thread("Server") {
+    this->port = port;
+    socket = new juce::StreamingSocket();
+    socket->bindToPort(port);
+    socket->createListener(port);
+    juce::Logger::getCurrentLogger()->writeToLog("Server startup");
+    running = true;

```

Comment:
Using Juce's `StreamingSocket` here to listen on a port, ready to accept TCP connections.